tutorials/cloudsdk-extensibility-type-safe-expand/cloudsdk-extensibility-type-safe-expand.md

```diff
@@ -0,0 +1,506 @@
+---
+title: Extensibility, Type-safe Expand, and Dependency Injection with the Virtual Data Model for OData
+description: The following tutorial will introduce you to the latest features of the SAP Cloud SDK regarding extensibility, eager and type-safe expand as well as dependency injection with the Virtual Data Model for OData for any SAP S/4HANA system.
+auto_validation: true
+time: 30
+tags: [ tutorial>intermediate, products>sap-s-4hana-cloud-sdk]
+primary_tag: topic>java
+---
+
+## Prerequisites
+ - [Introduce resilience to your application](https://developers.sap.com/tutorials/s4sdk-resilience.html)
+ - [Connect to OData Service on Cloud Foundry Using SAP Cloud SDK](s4sdk-odata-service-cloud-foundry)
+
+## Details
+### You will learn
+  - How you can use custom field extensions from S/4HANA within the virtual data model for OData.
+  - How you can join connected entities from the virtual data model in eager fashion.
+  - How to leverage dependency injection to decouple your client code better from the SDK-provided classes.
+
+---
+
+[ACCORDION-BEGIN [Step 1: ](Getting ready)]
+
+To successfully go through this tutorial, you will use the following `GetBusinessPartnerCommand` for all three features to the business partner API. The call itself requests the first and last name from all business partners which are customers.
+
+The file needs to be put under your `<projectroot>/application/src/main/java/com/sap/cloud/sdk/tutorial` directory:
```

Comment:
For all such cases throughout tutorial, you should say "The following file ..." or "This file ..."